mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
Oh, hm, maybe the idea is that checking this condition and 'fixing it' in the constructor is too costly. @jkbradley do I have the logic right?